controllers/storagecluster/cephcluster.go

```diff
@@ -349,6 +344,11 @@ func (obj *ocsCephCluster) ensureCreated(r *StorageClusterReconciler, sc *ocsv1.
 		}
 	}
 
+	// Set the default CephDeviceClass in the StorageCluster status if it's not set yet (not needed for external clusters)
+	if !sc.Spec.ExternalStorage.Enable && sc.Status.DefaultCephDeviceClass == "" && found.Status.CephStorage != nil {
```

Comment:
I thought ocs-operator hardcode the value somewhere for the device class that is set to storagecluster,
Can we match the list of device class from ceph with the default value, and set it in the status output

Reply:
Didn't fully understood, are you asking about setting the deviceclass on the osd's if they are not specified on storagecluster?

Reply:
So form 4.17 we always specify the device class, correct can we not just use that value from the storagecluster CR (if user has set or default value)

Reply:
@malayparida2000

Reply:
> form 4.17 we always specify the device class

I think that will be if they are created from the UI, we also need to think of what if the storagecluster is created from CLI without the deviceclass specified in the storage device set.

Reply:
got it, I thought it was from ocs operator